mx.sulong/mx_sulong_benchmarks.py

```diff
@@ -0,0 +1,257 @@
+        'gzip.c',
+        'nbody.cint.c',
+        'richards.c',
+        'fasta.cint.c',
+        'knucleotide.cint.c',
+        'oggenc.c',
+        'spectralnorm.cint.c'
+    ]
+
+def _benchmarksDirectory():
+    return join(os.path.abspath(join(mx.suite('sulong').dir, os.pardir)), 'sulong-benchmarks')
+
+def _configName():
+    if mx.suite('graal-enterprise', fatalIfMissing=False):
+        return 'enterprise'
+    elif mx.suite('graal-core', fatalIfMissing=False):
+        return 'core'
+    else:
+        return 'default'
+
+def _host_vm_tuple(args):
+    if mx.suite('graal-enterprise', fatalIfMissing=False):
+        hostVm = 'jvmci'
+        hostVmConfigPrefix = 'graal-enterprise'
+    elif mx.suite('graal-core', fatalIfMissing=False):
+        hostVm = 'jvmci'
+        hostVmConfigPrefix = 'graal-core'
+    else:
+        hostVm = 'server'
+        hostVmConfigPrefix = 'default'
+    return (hostVm, hostVmConfigPrefix)
+
+def _guest_vm_tuple(args):
+    return ('sulong', 'default')
+
+def _benchHigherScoreRegex():
+    return r'^(### )?(?P<benchmark>[a-zA-Z0-9\.\-_]+): +(?P<score>[0-9]+(?:\.[0-9]+)?)'
+
+def _failurePatterns():
+    return [
+        re.compile(r'error:'),
+        re.compile(r'Exception')
+    ]
+
+def _successPatterns():
+    return [re.compile(r'^(### )?([a-zA-Z0-9\.\-_]+): +([0-9]+(?:\.[0-9]+)?)', re.MULTILINE)]
+
+
+
+class NativeBenchmarkSuite(StdOutBenchmarkSuite):
+    def group(self):
+        return _group()
+
+    def subgroup(self):
+        return _subgroup()
+
+    def benchmarkList(self, bmSuiteArgs):
+        return _benchmarkList(bmSuiteArgs)
+
+    def benchHigherScoreRegex(self):
+        return _benchHigherScoreRegex()
+
+    def failurePatterns(self):
+        return _failurePatterns()
+
+    def successPatterns(self):
+        return _successPatterns()
+
+    def vmArgs(self, bmSuiteArgs):
+        return self.parseCommandLineArgs(bmSuiteArgs)[0]
+
+    def runArgs(self, bmSuiteArgs):
+        return self.parseCommandLineArgs(bmSuiteArgs)[1]
+
+    def parseCommandLineArgs(self, bmSuiteArgs):
+        vmArgs, runArgs = mx_benchmark.splitArgs(bmSuiteArgs, "--")
+
+        mx.logv('vmArgs: %s' % str(vmArgs))
+        mx.logv('runArgs: %s' % str(runArgs))
+
+        return vmArgs, runArgs
+
+    def before(self, bmSuiteArgs):
+        self.currentDir = os.getcwd()
+        os.chdir(_benchmarksDirectory())
+
+    def after(self, bmSuiteArgs):
+        os.chdir(self.currentDir)
+
+    def rules(self, out, benchmarks, bmSuiteArgs):
+        return [
+          mx_benchmark.StdOutRule(_benchHigherScoreRegex(), {
+                "vm": "sulong",
+                "benchmark": ("<benchmark>", str),
+                "metric.name": "score",
+                "metric.type": "numeric",
+                "metric.value": ("<score>", float),
+                "metric.score-function": "id",
+                "metric.better": "higher",
+                "metric.iteration": 0,
+                "host-vm": "native",
```

Comment:
So far we have the reverse convention: for VMs/executions where there is no concept of "host" vs. "guest" VM (e.g., GNU R), then it's the "guest" which is `none`.